ci/builder/Dockerfile

```diff
@@ -10,7 +10,66 @@
 # Build a cross-compiling toolchain that targets the oldest version of Linux
 # that we support.
 
-FROM ubuntu:jammy-20221130 AS crosstool
+FROM debian:bookworm-20230814 as crosstool
```

Comment:
What's the justification for changing this? We've been using Ubuntu successfully for years, and I'm loathe to switch something that's been working!

Reply:
Rather than using Ubuntu 23.04, which _isn't_ a long-term support release, I picked Debian 12 (which _is_) and is also consistent with what we're using in the cloud repo. Not wedded to this choice at all -- just wanted a base image that comes with Python 3.11, rather than having to pull it in externally!